scripts/us_census/pep/pep_by_srh/process.py

```diff
@@ -971,7 +1566,34 @@ def _consolidate_files():
     _consolidate_county_files()
     _consolidate_state_files()
     _consolidate_national_files()
-    _consolidate_all_geo_files()
+    _consolidate_all_geo_files(output_path)
+
+
+def add_future_year_urls():
+    global _FILES_TO_DOWNLOAD
+    with open(os.path.join(_MODULE_DIR, 'input_url.json'), 'r') as input_file:
+        _FILES_TO_DOWNLOAD = json.load(input_file)
+
+    urls_to_scan = [
+        "https://www2.census.gov/programs-surveys/popest/datasets/{YEAR}/counties/asrh/cc-est{YEAR}-alldata.csv"
+    ]
+
+    # This method will generate URLs for the years 2024 to 2029
+    for future_year in range(2024, 2030):
+        if dt.now().year > future_year:
+            YEAR = future_year
+            download_path = f"20{YEAR}_{YEAR+1}/county/cc-est{YEAR}-alldata.csv"  # Use f-string for dynamic path
```

Comment:
Should the base year be `/2020_2023/` instead of `/2022_2023/`?

Reply:
Updated.